templates/ode.py.jinja

```diff
@@ -90,99 +90,4 @@ def system(t: np.float64, y: np.ndarray, *constants) -> np.ndarray:
 
-
-def simulation_output_to_csv(sim_steps, simulation_output, write_to) -> str:
-    if not simulation_output.success:
-        print(simulation_output.message)
-        return
-
-    populatio_values_per_dt = simulation_output.y.T
-
-    write_to.write(f"t,{','.join(variable_names())}\n")
-
-    for dt, y in zip(sim_steps, populatio_values_per_dt):
-        write_to.write(f"{dt},")
-        write_to.write(",".join(f"{val:.4f}" for val in y))
-        write_to.write("\n")
-
-
-def plot_simulation(sim_steps, simulation_output, filename):
-    import matplotlib.pyplot as plt
-    from matplotlib.backends.backend_pdf import PdfPages
-
-    with PdfPages(filename) as pdf:
-        # All
-        all_fig, all_ax = plt.subplots()
-        all_fig.set_size_inches(8, 6)
-        all_ax.set(title="", xlabel="Time", ylabel="Concentration")
-
-        # Individually
-        for variable_name, variable_line_data in zip(variable_names(), simulation_output.y):
-            fig, ax = plt.subplots()
-            fig.set_size_inches(8, 6)
-            ax.set(
-                title=variable_name,
-                xlabel="Time",
-                ylabel="Concentration",
-            )            
-            ax.plot(simulation_output.t, variable_line_data)
-            all_ax.plot(simulation_output.t, variable_line_data)
-
-            pdf.savefig(fig)
-        all_ax.legend(variable_names(),loc="best")
-        pdf.savefig(all_fig)
-
-
-def file_or_stdout(filename: str | None):
-    if filename:
-        return open(filename, 'w')
-    else:
-        return sys.stdout
-
-
-def simulate(filename, st=0, tf=50, dt=0.1, plot=False):
-    sim_steps = np.arange(st, tf + dt, dt)
-
-    simulation_output = scipy.integrate.solve_ivp(
-        fun=system,
-        t_span=(st, tf + dt * 2),
-        y0=initial_values(),
-        args=constants(),
-        t_eval=sim_steps,
-    )
-
-    if plot:
-        plot_simulation(sim_steps, simulation_output, filename)
-
-    else:
-        with file_or_stdout(filename) as f:
-            simulation_output_to_csv(sim_steps, simulation_output, f)
-
-
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-
-    parser.add_argument("--st", type=float, default=0)
-    parser.add_argument("--tf", type=float, default=50)
-    parser.add_argument("--dt", type=float, default=0.01)
-    parser.add_argument("-o", "--output", default=None)
-    parser.add_argument("--csv", action=argparse.BooleanOptionalAction)
-
-    args = parser.parse_args()
-
-    if args.output is None and not args.csv:
-        parser.error("when plotting (a.k.a --no-csv), an output file name is required via --output")
-
-    if args.output:
-        dirs = os.path.dirname(args.output)
-
-        if dirs:
-            os.makedirs(dirs, exist_ok=True)
-
-    simulate(
-        args.output,
-        plot=not args.csv,
-        st=args.st,
-        tf=args.tf,
-        dt=args.dt,
-    )
+# includes! "ode-support.py"
```

Comment:
Só uma dúvida, mas essa linha é algum controle especial do Jinja / `include_str!` ?

Reply:
Não, essa linha é pra indicar para alguém que for editar no futuro que esse arquivo será concatenado no final.

Eu gostaria de utilizar o comando `{% - include %}` do jinja, mas a gente não tem acesso aos templates em tempo de execução, então fica ai essa solução "humanizada" kk